ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/commands/tracker/CommandTracker.scala

```diff
@@ -225,10 +226,19 @@ private[commands] class CommandTracker[Context](maxDeduplicationTime: () => JDur
             }
             val commandTimeout = {
               lazy val maxDedup = maxDeduplicationTime()
-              val dedup = commands.deduplicationTime.getOrElse(
-                ProtoDuration.of(maxDedup.getSeconds, maxDedup.getNano)
-              )
-              Instant.now().plusSeconds(dedup.seconds).plusNanos(dedup.nanos.toLong)
+              lazy val protoMaxDedup = ProtoDuration.of(maxDedup.getSeconds, maxDedup.getNano)
+              val timeoutDuration = commands.deduplication match {
+                case Deduplication.Empty =>
+                  protoMaxDedup
+                case Deduplication.DeduplicationTime(value) =>
+                  value
```

Comment:
If you converted this to a `Duration` here, it would simplify the rest of the function.